### ISSUE TITLE
How to save a simple logistic regression model?

### ISSUE THREAD

**Issue by minasmz** [ratschlab/HIRID-ICU-Benchmark#7]
Hi,

I was trying to build a logistic regression model on the preprocessed data. The model is built and test and validation .pkl files are created in the log folder. I tried to save the logistic regression model as a .pkl file as well. As a variable (save_weights) defined in the train function of the wrappers.py file for this and I set it to "True" to save the model as follow:

@gin.configurable(module='MLWrapper')
    def train(self, train_dataset, val_dataset, weight,
              patience=gin.REQUIRED, save_weights=True):

But I get the error of:
_pickle.PicklingError: Can't pickle <class 'sklearn.linear_model._logistic.LogisticRegression'>: it's not the same object as sklearn.linear_model._logistic.LogisticRegression
  In call to configurable 'train' (<function MLWrapper.train at 0x141d17670>)
  In call to configurable 'train_common' (<function train_common at 0x1414d1670>)

and I cannot find the issue. It seems that the LogisticRegression functions are different...
I appreciate it if you help me to find the problem and solve this issue.

**Comment by hugoych** [comment]
Hi, 
This is an issue we also encountered at some point. The reason is a conflict between `gin-config` and `pickle` making configurable objects no longer pickable (see issue [#31 of gin-config ](https://github.com/google/gin-config/issues/31).


The issue has been solved in `gin-config` but not in the latest official release `0.4.0`. Thus we require to use a specific version of gin including the fix for issue #31 (see [`enviromnent.yml`](https://github.com/ratschlab/projects2021-hirid-icu-benchmark-internal/blob/8bb039819e0693fa44d7fcb5d3a27097e843b3b2/environment.yml#L27)).

What version of gin-config are you using?
